### PR TITLE
Build Lighthouse images for x86-64 and ARM64

### PR DESCRIPTION
**Pull request by skitt**
This relies on cross-building support in the Shipyard base image and
project Makefiles to avoid having to provide a native Shipyard base
image (and slower compilation using QEMU). The final image build stage
relies on support for the target architecture however.

The TARGETPLATFORM default is dropped from the container definitions;
it is expected to be provided by the build environment (buildx
currently), and setting it prevents us from actually building foreign
images.

Depends on https://github.com/submariner-io/lighthouse/pull/683
Depends on https://github.com/submariner-io/shipyard/pull/721
Depends on https://github.com/submariner-io/shipyard/pull/728
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by dfarrell07 — DISMISSED** [review]

**Review by sridhargaddam — DISMISSED** [review]

**Comment by skitt** [comment]
I’m dismissing the reviews since reviews on PRs on the backlog don’t make much sense.

File: Makefile
Comment (mkolesnik) [inline Makefile:9]: This is already defined is Submariner's `Makefile` and I suspect would be needed in other projects (operator?)

Perhaps it's best to defined these functions in `Makefile.inc`?

File: Makefile
Comment (mkolesnik) [inline Makefile:13]: This too :) ?

File: Makefile
Comment (mkolesnik) [inline Makefile:46]: I wonder if it's really necessary to do it for all the binaries, and not just one arch (shouldnt they have similar license regardless of platform?)

File: package/Dockerfile.lighthouse-agent
Comment (mkolesnik) [inline package/Dockerfile.lighthouse-agent:12]: This seems redundant per [this blog post]?

> For this stage, we will not use --platform in the FROM definition. We could write FROM --platform=$TARGETPLATFORM but that is the default value for all builds stages anyway so using a flag is redundant.

[this blog post]: https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

File: Makefile
Comment (skitt) [reply Makefile:9]: Yes, but I needed them here to have a working PR ;-). I’ll move everything appropriate to Shipyard once this gets merged.

File: Makefile
Comment (skitt) [reply Makefile:46]: Some platforms can result in different dependencies (_e.g._ Windows).

File: package/Dockerfile.lighthouse-agent
Comment (skitt) [reply package/Dockerfile.lighthouse-agent:12]: It’s redundant per the official docs too, but I included it for balance. I thought it preferable to be explicit, so that readers wouldn’t have to wonder what the platform is when it’s not specified.

**Comment by skitt** [comment]
The UPX logs in the Multi-arch job show that this is building amd64 and arm64 binaries and images.

File: .github/workflows/multiarch.yml
Comment (dfarrell07) [inline .github/workflows/multiarch.yml]:
```suggestion
name: Multi-arch Builds
```

**Review by tpantelis — APPROVED** [review]

File: .github/workflows/multiarch.yml
Comment (dfarrell07) [reply .github/workflows/multiarch.yml]: We can go ahead and merge this, fix the name in a follow-up commit as was done in other repos.

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**
